indico/modules/rb_new/client/js/common/bookings/BookingObjectLink.jsx

```diff
@@ -0,0 +1,117 @@
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Message, Placeholder, Segment, Icon} from 'semantic-ui-react';
+import getLinkedObjectDataURL from 'indico-url:rooms_new.linked_object_data';
+import {Translate, Param} from 'indico/react/i18n';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import camelizeKeys from 'indico/utils/camelize';
+
+import './BookingObjectLink.module.scss';
+
+/**
+ * `BookingObjectLink` displays a message informing if the booking is
+ * linked or will be linked to an event, contribution or session block.
+ */
+export default class BookingObjectLink extends React.PureComponent {
+    static propTypes = {
+        type: PropTypes.oneOf(['event', 'contrib', 'sessionBlock']).isRequired,
+        id: PropTypes.number.isRequired,
+        /** Whether it is a potential link or the booking is already linked */
+        potential: PropTypes.bool,
+    };
+
+    static defaultProps = {
+        potential: false
+    };
+
+    state = {
+        loaded: false,
+        data: {},
+    };
+
+    componentDidMount() {
+        const {type, id} = this.props;
+        this.fetchLinkedObjectData(type, id);
+    }
+
+    async fetchLinkedObjectData(type, id) {
+        let response;
+        type = _.snakeCase(type);
+        try {
+            response = await indicoAxios.get(getLinkedObjectDataURL({type, id}));
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        this.setState({
+            loaded: true,
+            data: camelizeKeys(response.data)
+        });
+    }
+
+    renderObjectInfo(data) {
+        const {type, potential} = this.props;
+        const {url, title, canAccess, eventTitle, eventUrl} = data;
+        const mapping = {
+            event: Translate.string('an event'),
+            contrib: Translate.string('a contribution'),
+            sessionBlock: Translate.string('a session block')
+        };
+        const verb = potential ? <Translate> will be </Translate> : <Translate> is </Translate>;
+        return !!mapping[type] && canAccess && (
+            <Message icon color="teal">
+                <Icon name="linkify" />
+                <Message.Content>
+                    <>
+                        <Translate>
+                            This booking <Param name="verb" value={verb} /> linked to <Param name="type" wrapper={<strong />}
```

Comment:
this can be wrapped in a more readable way, even without requiring `{' '}` to keep proper spacing:
```jsx
                            This booking <Param name="verb" value={verb} /> linked
                            to <Param name="type" wrapper={<strong />} value={mapping[type]} />:
```